### PR TITLE
[Typing][PEP585 Upgrade][BUAA][111-120] Use standard collections for type hints for 10 files in `python/paddle/`

### PR DESCRIPTION
**Pull request by Whsjrczr**
### PR Category

User Experience 

### PR Types

Others

### Description

python/paddle/quantization/config.py
python/paddle/reader/decorator.py
python/paddle/sparse/nn/functional/conv.py
python/paddle/sparse/nn/layer/conv.py
python/paddle/sparse/unary.py
python/paddle/tensor/array.py
python/paddle/tensor/creation.py
python/paddle/tensor/einsum.py
python/paddle/tensor/linalg.py
python/paddle/tensor/manipulation.py

### CODE REVIEW

**Comment by Whsjrczr** [comment]
请问Generator的import时的位置该放在哪里

File: python/paddle/reader/decorator.py [hunk 1]
```diff
@@ -20,20 +20,22 @@
...
-    Generator,
-    Sequence,
     TypedDict,
     TypeVar,
     overload,
 )
 
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+
```
Comment (gouzil) [inline python/paddle/reader/decorator.py]: 这为啥又来一次`TYPE_CHECKING`

File: python/paddle/sparse/nn/functional/conv.py [hunk 2]
```diff
@@ -14,7 +14,10 @@
 
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, Literal, Sequence
+from typing import TYPE_CHECKING, Literal
+
+if TYPE_CHECKING:
```
Comment (gouzil) [inline python/paddle/sparse/nn/functional/conv.py]: 同上，后面的就不一一评论了

**Comment by gouzil** [comment]
> 请问Generator的import时的位置该放在哪里

现在的位置就是对的

File: python/paddle/sparse/nn/functional/conv.py (hunk 2 above)
Comment (Whsjrczr) [reply python/paddle/sparse/nn/functional/conv.py]: 已更新

File: python/paddle/reader/decorator.py [hunk 3]
```diff
@@ -20,15 +20,14 @@
 import random
 import sys
 import warnings
+from collections.abc import Generator
```
Comment (SigureMo) [inline python/paddle/reader/decorator.py]: 这个在 type alias 里用了，不能改，运行时会挂

Comment (Whsjrczr) [reply python/paddle/reader/decorator.py]: 收到！

**Comment by Whsjrczr** [comment]
PR-CI-SOT 还是过不了，报错：
2024-08-09 13:27:00   File "/workspace/Paddle/build/python/paddle/reader/decorator.py", line 72, in <module>
2024-08-09 13:27:00     _Reader: TypeAlias = Callable[[], Generator[_T, None, None]]
2024-08-09 13:27:00 TypeError: 'ABCMeta' object is not subscriptable
请问是Callable位置不对嘛

**Review by SigureMo — APPROVED** [review]
LGTMeow

**Pull request merged**